dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt

```diff
@@ -0,0 +1,181 @@
+         * This feature is enabled by default, disabling it will prevent any crash report to be
+         * sent to Datadog servers.
+         * @param enabled true by default
+         */
+        fun setCrashReportsEnabled(enabled: Boolean): Builder {
+            crashReportsEnabled = enabled
+            return this
+        }
+
+        /**
+         * Sets the service name that will appear in your logs.
+         * @param serviceName the service name (default = "android")
+         */
+        fun setServiceName(serviceName: String): Builder {
+            logsConfig = logsConfig.copy(serviceName = serviceName)
+            tracesConfig = tracesConfig.copy(serviceName = serviceName)
+            crashReportConfig = crashReportConfig.copy(serviceName = serviceName)
+            return this
+        }
+
+        /**
+         * Let the SDK target Datadog's Europe server.
+         *
+         * Call this if you log on [app.datadoghq.eu](https://app.datadoghq.eu/).
+         */
+        fun europeEndpoints(): Builder {
+            logsConfig = logsConfig.copy(endpointUrl = DatadogEndpoint.LOGS_EU)
+            tracesConfig = tracesConfig.copy(endpointUrl = DatadogEndpoint.TRACES_EU)
+            crashReportConfig = crashReportConfig.copy(endpointUrl = DatadogEndpoint.LOGS_EU)
+            needsClearTextHttp = false
+            return this
+        }
+
+        /**
+         * Let the SDK target Datadog's US server.
+         *
+         * Call this if you log on [app.datadoghq.com](https://app.datadoghq.com/).
+         */
+        fun usEndpoints(): Builder {
+            logsConfig = logsConfig.copy(endpointUrl = DatadogEndpoint.LOGS_US)
+            tracesConfig = tracesConfig.copy(endpointUrl = DatadogEndpoint.TRACES_US)
+            crashReportConfig = crashReportConfig.copy(endpointUrl = DatadogEndpoint.LOGS_US)
+            needsClearTextHttp = false
+            return this
+        }
+
+        /**
+         * Let the SDK target a custom server for the logs feature.
+         */
+        fun customLogsEndpoint(endpoint: String): Builder {
+            logsConfig = logsConfig.copy(endpointUrl = endpoint)
+            checkCustomEndpoint(endpoint)
+            return this
+        }
+
+        /**
+         * Let the SDK target a custom server for the tracing feature.
+         */
+        fun customTracesEndpoint(endpoint: String): Builder {
+            tracesConfig = tracesConfig.copy(endpointUrl = endpoint)
+            checkCustomEndpoint(endpoint)
+            return this
+        }
+
+        /**
+         * Let the SDK target a custom server for the crash reports feature.
+         */
+        fun customCrashReportsEndpoint(endpoint: String): Builder {
+            crashReportConfig = crashReportConfig.copy(endpointUrl = endpoint)
+            checkCustomEndpoint(endpoint)
+            return this
+        }
+
+        private fun checkCustomEndpoint(endpoint: String) {
```

Comment:
Because we have only one shared OkHttp instance, we need it to have max compatibility with all cases, so it's a global thing. 
Besides, the clear text is in theory only necessary for our own instrumented tests.